tests/Transports.AspNetCore.Tests/Middleware/BatchTests.cs

```diff
@@ -0,0 +1,248 @@
+        using var response = await PostBatchRequestAsync(new GraphQLRequest() { Query = "{" });
+        // validation errors do not return 400 within a batch request
+        await response.ShouldBeAsync(false, @"[{""errors"":[{""message"":""Error parsing query: Expected Name, found EOF; for more information see http://spec.graphql.org/October2021/#Field"",""locations"":[{""line"":1,""column"":2}],""extensions"":{""code"":""SYNTAX_ERROR"",""codes"":[""SYNTAX_ERROR""]}}]}]");
+    }
+
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task NoQuery(bool badRequest)
+    {
+        _options.ValidationErrorsReturnBadRequest = badRequest;
+        using var response = await PostJsonAsync("[{}]");
+        // validation errors do not return 400 within a batch request
+        await response.ShouldBeAsync(false, @"[{""errors"":[{""message"":""GraphQL query is missing."",""extensions"":{""code"":""QUERY_MISSING"",""codes"":[""QUERY_MISSING""]}}]}]");
+    }
+
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task NoQuery_Multiple(bool badRequest)
+    {
+        _options.ValidationErrorsReturnBadRequest = badRequest;
+        using var response = await PostJsonAsync("[{},{}]");
+        // validation errors do not return 400 within a batch request
+        await response.ShouldBeAsync(false, @"[{""errors"":[{""message"":""GraphQL query is missing."",""extensions"":{""code"":""QUERY_MISSING"",""codes"":[""QUERY_MISSING""]}}]},{""errors"":[{""message"":""GraphQL query is missing."",""extensions"":{""code"":""QUERY_MISSING"",""codes"":[""QUERY_MISSING""]}}]}]");
+    }
+
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task NullRequest(bool badRequest)
+    {
+        _options.ValidationErrorsReturnBadRequest = badRequest;
+        using var response = await PostJsonAsync("[null]");
+        // validation errors do not return 400 within a batch request
+        await response.ShouldBeAsync(false, @"[{""errors"":[{""message"":""GraphQL query is missing."",""extensions"":{""code"":""QUERY_MISSING"",""codes"":[""QUERY_MISSING""]}}]}]");
+    }
+
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task NullRequest_Multiple(bool badRequest)
+    {
+        _options.ValidationErrorsReturnBadRequest = badRequest;
+        using var response = await PostJsonAsync("[null,null]");
+        // validation errors do not return 400 within a batch request
+        await response.ShouldBeAsync(false, @"[{""errors"":[{""message"":""GraphQL query is missing."",""extensions"":{""code"":""QUERY_MISSING"",""codes"":[""QUERY_MISSING""]}}]},{""errors"":[{""message"":""GraphQL query is missing."",""extensions"":{""code"":""QUERY_MISSING"",""codes"":[""QUERY_MISSING""]}}]}]");
+    }
+
+    [Fact]
+    public async Task Mutation()
+    {
+        using var response = await PostBatchRequestAsync(new GraphQLRequest() { Query = "mutation{clearMessages}" });
```

Comment:
```suggestion
        using var response = await PostBatchRequestAsync(new GraphQLRequest { Query = "mutation{clearMessages}" });
```

and for other usages

Reply:
I agree and will accept suggestions where you make them along these lines.  I'm not searching the tests to change them.